compiler/data/var-data.h

```diff
@@ -32,6 +32,7 @@ class VarData {
   std::string name;
   tinf::VarNode tinf_node;
   VertexPtr init_val;
+  bool had_user_assigned_val = false;
```

Comment:
Wouldn't `uninited_flag` be enough?
I had a quick look and I suppose it's what you need

Reply:
No, it is not.
For this specific case it is important to know if user has provided value, this is the core root of a problem here.

Reply:
But `uninited_flag` is what we need. In always `true` if there was a provided `init_val` (even if it was optimized out) as I understand, isn't it?

Reply:
I checked your proposal, no, it is not enough for this specific case. Value is wiped out and we get error. Let leave this case as is.